### PR TITLE
Use Boolean Hamiltonian gates for QAOA example

### PR DESCRIPTION
**Pull request by tonybruguier**

### CODE REVIEW

File: examples/examples_test.py [hunk 1]
```diff
@@ -93,7 +93,8 @@ def test_example_heatmaps():
 
 
 def test_example_runs_qaoa():
-    examples.qaoa.main(repetitions=10, maxiter=5)
+    examples.qaoa.main(repetitions=10, maxiter=5, use_boolean_hamiltonian_gate=False)
+    examples.qaoa.main(repetitions=10, maxiter=5, use_boolean_hamiltonian_gate=True)
```
Comment (dabacon) [inline examples/examples_test.py:98]: maybe add a test that the circuits produced for given parameters end up with the same unitary up to a global phase?

**Review by dabacon — APPROVED** [review]
One optional suggestion for testing.

File: examples/examples_test.py (hunk 1 above)
Comment (tonybruguier) [reply examples/examples_test.py:98]: Done.

**Comment by tonybruguier** [comment]
Thanks, @dabacon I added the unit test you requested.

PTAL.

**Review by vtomole — APPROVED** [review]
Thanks

**Pull request merged**